cms/static/js/views/settings/main.js

```diff
@@ -26,6 +26,12 @@ var DetailsView = ValidatingView.extend({
         this.$el.find("#course-organization").val(this.model.get('org'));
         this.$el.find("#course-number").val(this.model.get('course_id'));
         this.$el.find("#course-name").val(this.model.get('run'));
+
+        if(!this.model.has('license')){
+            this.model.set('license', {kind: 'ARR', version: '1.0'});
+        }
+        this.linkLicense();
```

Comment:
Why don't you have the `linkLicense` method listen to the model, and call itself automatically when the `license` attribute changes?